os/binfmt/binfmt_loadbinary.c

```diff
@@ -204,12 +208,30 @@ int load_binary(int binary_idx, FAR const char *filename, load_attr_t *load_attr
 			goto errout_with_unload;
 		}
 
-		return pid;
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
 	}
 
-	return OK;
+	pid = OK;
```

Comment:
This will overwrite the pid value from line 204.
Better to initialize pid = 0 at line 191 or at the time of declaration.